Mail/Views/Thread/Message/MessageBannerHeaderView.swift

```diff
@@ -27,56 +28,55 @@ struct MessageBannerHeaderView: View {
 
     @State private var isButtonLoading = false
 
+    let banners: [MessageBanner]
+
     @ObservedRealmObject var message: Message
     @ObservedRealmObject var mailbox: Mailbox
 
-    @Binding var displayContentBlockedActionView: Bool
-
-    private var isRemoteContentBlocked: Bool {
-        return (UserDefaults.shared.displayExternalContent == .askMe || message.folder?.role == .spam)
-            && !message.localSafeDisplay
-    }
-
     var body: some View {
-        let spamType = spamTypeFor(message: message)
-        if let spamType {
-            MessageHeaderActionView(
-                icon: spamType.icon,
-                message: spamType.message,
-                isFirst: true
-            ) {
-                Button {
-                    action()
-                } label: {
-                    Text(spamType.buttonTitle)
+        ForEach(banners, id: \.id) { banner in
```

Comment:
Is it really needed ?
```suggestion
        ForEach(banners) { banner in
```